chirp/drivers/kguv980p.py

```diff
@@ -2460,28 +2424,55 @@ def _decode(lst):
                            RC_POWER_LIST[_settings.rc_power]))
         remote_grp.append(rs)
 
-        _code = ''
-        test = int(_settings.ani_edit)
-        _codeobj = '0x{0:0{1}X}'.format(test, 6)
-        LOG.debug("codeobj = %s" % _codeobj)
-        _ani = str(_codeobj)
-        for i in range(2, 8):
-            LOG.debug("ani[i] = %s" % _ani[i])
-            if _ani[i] in dtmfchars:
-                _code += _ani[i]
-        val_ani = int(_code)
+        def decode_remote_vals(self, setting):
+            _code = ''
+            # convert the stored 24bit decimal value to 6 individual
+            # digits in hex format into a string.
+            # in the radio an id of 000000 is stored as 0x000000
+            # a id value of 101 is stored as 0x101CCC
+            # pad the value with up to 6 zeroes to handle
+            # default value of 000000 and not simply display 0
+            value = int(setting)
+            _codeobj = "{0:0{1}X}".format(value, 6)
+            LOG.debug("codeobj = %s" % _codeobj)
+            _val = _codeobj
+            # parse each char in _codeobj and only add to
+            # _code if it is a valid DTMF char
+            # for a value of 0x101CCC we only want to display 101
+            for i in range(0, 6):
+                LOG.debug("val [i] = %s" % _val[i])
+                if _val[i] in dtmfchars:
+                    _code += _val[i]
+            val_out = int(_code)
```

Comment:
It still seems like there has to be a much less convoluted way to do this than stringifying it. But even if you go that route, I think all this (from L2428 to L2447 can be replaced with:
```
code = ('%06X' % int(setting)).replace('C' '')
return int(code), code
```
...right? It's just much easier to grok.

Reply:
that did work and is alot easier.